File: src/main/java/org/vsegda/servlet/AlertServlet.java
package org.vsegda.servlet;

import org.vsegda.util.Alert;

import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.io.IOException;

/**
 * @author Roman Elizarov
 */
public class AlertServlet extends HttpServlet {
	@Override
	protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
		String text = req.getParameter("text");
		if (text == null)
			throw new ServletException("text parameter required");
		String id = req.getParameter("id");
		Alert.sendAlertEmail(id, text);
		
	}
}

File: service/src/main/java/org/vsegda/servlet/AlertServlet.kt
package org.vsegda.servlet

import org.vsegda.service.*
import java.io.*
import javax.servlet.*
import javax.servlet.http.*

class AlertServlet : HttpServlet() {
    override fun doPost(req: HttpServletRequest, resp: HttpServletResponse) {
        val text = req.getParameter("text") ?: throw ServletException("text parameter required")
        val id = req.getParameter("id")
        AlertService.sendAlertEmail(id, text)
    }
}
